hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -182,15 +192,23 @@ long uploadJobResources(JobConfig jobConfig) {
         try {
             for (ResourceConfig rc : jobConfig.getResourceConfigs()) {
                 switch (rc.getResourceType()) {
-                    case REGULAR_FILE:
-                        InputStream in = rc.getUrl().openStream();
-                        readStreamAndPutCompressedToMap(rc.getId(), tmpMap, in);
-                        in.close();
+                    case CLASS:
+                        try (InputStream in = rc.getUrl().openStream()) {
+                            readStreamAndPutCompressedToMap(rc.getId(), tmpMap, in);
+                        }
+                        break;
+                    case FILE:
+                        IMapOutputStream os = new IMapOutputStream(getJobFileStorage(jobId).get(), rc.getId());
```

Comment:
This is broken, we get a lazy concurrent supplier each time and immediately do a `get()` on it. The `getJobFileStorage(jobId)` should be stored outside of the loop. And it doesn't have to be concurrent.